### PR TITLE
[cpackget] update-index: added option "-a" to download all missing PDSC files that are listed in index.pidx

### PR DESCRIPTION
**Pull request by thorstendb-ARM**
Command "update-index":
- added "--all-pdsc-files", "-a"
- downloads all PDSC files that are listed in index.pidx but are not present in .Web
- if "sparse" is not set, the files that have already been in .Web are updated, if necessary (newly downloaded files are not checked)

### CODE REVIEW

File: cmd/commands/update_index.go [hunk 1]
```diff
@@ -46,4 +48,5 @@ By default it will also check if all PDSC files under .Web/ need update as well.
 
 func init() {
 	UpdateIndexCmd.Flags().BoolVarP(&updateIndexCmdFlags.sparse, "sparse", "s", false, "avoid updating the pdsc files within .Web/ folder")
+	UpdateIndexCmd.Flags().BoolVarP(&updateIndexCmdFlags.downloadUpdatePdscFiles, "all-pdsc-files", "a", false, "updates / downloads all the latest .pdsc files from the public index")
```
Comment (soumeh01) [inline cmd/commands/update_index.go:51]:
```suggestion
	UpdateIndexCmd.Flags().BoolVarP(&updateIndexCmdFlags.downloadUpdatePdscFiles, "all-pdsc-files", "a", false, "updates/downloads all the latest .pdsc files from the public index")
```

File: cmd/installer/root_test.go [hunk 2]
```diff
@@ -416,7 +416,7 @@ func TestUpdatePublicIndex(t *testing.T) {
 
 		indexPath := server.URL() + "this-file-does-not-exist"
 
-		err := installer.UpdatePublicIndex(indexPath, Overwrite, Sparse, DownloadPdsc, Concurrency, Timeout)
+		err := installer.UpdatePublicIndex(indexPath, Overwrite, Sparse, DownloadPdsc, false, Concurrency, Timeout)
```
Comment (soumeh01) [inline cmd/installer/root_test.go:419]: Where can we find the test case for 'true' case?

Comment (thorstendb-ARM) [reply cmd/installer/root_test.go]: see "test add remaining PDSC from index.pidx"

**Review by soumeh01 — APPROVED** [review]

**Review by soumeh01 — APPROVED** [review]

**Review by jkrech — APPROVED** [review]
Looks good to me.

**Pull request merged**